ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/config/CliSpec.scala

```diff
@@ -0,0 +1,300 @@
+          beginOffset = None,
+          endOffset = None,
+          objectives = None,
+        ),
+        s"stream-type=active-contracts,name=$name,filters=$party" -> ActiveContractsStreamConfig(
+          name = name,
+          filters = List(PartyFilter(party, Nil)),
+          objectives = None,
+        ),
+        s"stream-type=completions,name=$name,party=$party,application-id=$appId" -> CompletionsStreamConfig(
+          name = name,
+          party = party,
+          applicationId = appId,
+          beginOffset = None,
+          objectives = None,
+        ),
+      )
+      forAll(cases) { (argument, config) =>
+        val expectedConfig =
+          Config.Default.copy(workflow = Config.Default.workflow.copy(streams = List(config)))
+        parse("--consume-stream", argument) shouldBe expectedConfig
+        parse("-s", argument) shouldBe expectedConfig
+      }
+    }
+
+    "parse stream filters" in {
+      import WorkflowConfig.StreamConfig._
+      val name = "streamname"
+      val party1 = "alice"
+      val party2 = "bob"
+      val party3 = "david"
+      val template1 = "packageid:Foo:Foo1"
+      val template2 = "packageid2:Foo:Foo2"
+      // each party filter separated by '+' and each template in a filter separated by '@'
+      val filters = s"$party1+$party2@$template1@$template2+$party3@$template2"
+      val filtersList = List(
+        PartyFilter(party1, List()),
+        PartyFilter(party2, List(template1, template2)),
+        PartyFilter(party3, List(template2)),
+      )
+      val cases = Table(
+        "cli argument" -> "stream config",
+        s"stream-type=transactions,name=$name,filters=$filters" -> TransactionsStreamConfig(
+          name = name,
+          filters = filtersList,
+          beginOffset = None,
+          endOffset = None,
+          objectives = None,
+        ),
+        s"stream-type=transaction-trees,name=$name,filters=$filters" -> TransactionTreesStreamConfig(
+          name = name,
+          filters = filtersList,
+          beginOffset = None,
+          endOffset = None,
+          objectives = None,
+        ),
+        s"stream-type=active-contracts,name=$name,filters=$filters" -> ActiveContractsStreamConfig(
+          name = name,
+          filters = filtersList,
+          objectives = None,
+        ),
+      )
+      forAll(cases) { (argument, config) =>
+        val expectedConfig =
+          Config.Default.copy(workflow = Config.Default.workflow.copy(streams = List(config)))
+        parse("--consume-stream", argument) shouldBe expectedConfig
+        parse("-s", argument) shouldBe expectedConfig
+      }
+    }
+
+    "parse begin offset" in {
+      import WorkflowConfig.StreamConfig._
+      val name = "streamname"
+      val party = "dummy"
+      val cases = Table(
+        "cli parameter" -> "offset",
+        "abcdef" -> LedgerOffset.defaultInstance.withAbsolute("abcdef"),
+        "ledger-begin" -> LedgerOffset.defaultInstance.withBoundary(
```

Comment:
It is expected that CLI and YAML parsers will accept such configuration because I don't want to put too much domain logic in there. I've added unit tests that check if such parsing is possible.
We may consider blocking such configuration at the service level but at the moment such configuration will just produce a request with empty results.